File: LeetCode_Marathons/LCM4/Medium/Coin_Change_II/a.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        dp: List[int] = [0] * (amount + 1)
        dp[0] = 1
        for coinAmount in coins:
            for i in range(1, amount + 1):
                if i - coinAmount >= 0:
                    dp[i] += dp[i - coinAmount]
        return dp[amount]

File: LeetCode_Marathons/LCM4/Medium/Coin_Change_II/a.java
class Solution {
    public int change(int amount, int[] coins) {
        int[] dp = new int[amount + 1];
        dp[0] = 1;
        for (int coinAmount : coins) {
            for (int i = 0; i <= amount; i++) {
                if (i - coinAmount >= 0) {
                    dp[i] += dp[i - coinAmount];
                }
            }
        }
        return dp[amount];
    }
}